engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -1976,7 +1979,7 @@ namespace dmGameObject
         dmHashRelease64(&instance->m_CollectionPathHashState);
         if(instance->m_Generated)
         {
-            dmHashReverseErase64(instance->m_Identifier);
+            dmScript::ReleaseHash(dmScript::GetLuaState(g_ScriptContext), instance->m_Identifier);
```

Comment:
Delete hash on deletion of all dynamically generated game objects (spawned with a factory)